pkg/vsphere/disk/disk.go

```diff
@@ -45,6 +45,43 @@ type Filesystem interface {
 	Unmount(path string) error
 }
 
+// ReferenceCount represents the number of references to a disk
+type ReferenceCount struct {
+	owner   string
+	refname string
+	count   int
+}
+
+// NewReferenceCount creates and returns a ReferenceCount initialized to 0
+func NewReferenceCount(o, n string) *ReferenceCount {
+	return &ReferenceCount{
+		owner:   o,
+		refname: n,
+		count:   0,
+	}
+}
+
+// Increment increases the reference count by one
+func (r *ReferenceCount) Increment() {
+	r.count++
```

Comment:
There are already locks around this code albeit at a higher level. Are you sure you want to lock within a lock?